pilot/pkg/serviceregistry/kube/controller/controller.go

```diff
@@ -738,6 +639,46 @@ func (c *Controller) getProxyServiceInstancesByEndpoint(endpoints v1.Endpoints,
 	return out
 }
 
+func (c *Controller) getProxyServiceInstancesByEndpointSlice(slice discoveryv1alpha1.EndpointSlice, proxy *model.Proxy) []*model.ServiceInstance {
```

Comment:
nit: seems like this should be part of the endpointslicecontroller.  It's a bit of a chicken-and-the-egg problem because then the ESC would need a reference back to the controller, but it think it would help to better separate the endpointslice-specific logic.

Reply:
+1, this should be the implement detail

Reply:
I moved this to endpointslice, and did the same for the Endpoints one